### ISSUE TITLE
semi-supervised noisy label augmentation

### ISSUE THREAD

**Issue by dmarx** [dmarx/SuicideWatch#8]
* http://ruder.io/semi-supervised/ 
* [Training Deep Neural Networks on Noisy Labels with Bootstrapping](https://arxiv.org/abs/1412.6596)
* [Training Convolutional Networks with Noisy Labels](https://arxiv.org/abs/1406.2080)
* [Learning from noisy labels with dropout regularization](https://arxiv.org/abs/1705.03419)
* [Learning from noisy labels with deep neural networks](http://www-labs.iro.umontreal.ca/~bengioy/cifar/NCAP2014-summerschool/slides/Sainbayar_cifar_sainaa_compressed.pdf)
* [Dimensionality driven learning with noisy labels](http://proceedings.mlr.press/v80/ma18d/ma18d.pdf)
* [Toward Robustness against Label Noise in
Training Deep Discriminative Neural Networks](https://papers.nips.cc/paper/7143-toward-robustness-against-label-noise-in-training-deep-discriminative-neural-networks.pdf)
* [Learning with Noisy Labels](https://papers.nips.cc/paper/5073-learning-with-noisy-labels.pdf)
* [Iterative learning with open-set noisy labels](http://wyliu.com/papers/LiuCVPR18_ONL.pdf) - siamese network

PU Learning: positive + unlabeled data
* https://www.cs.uic.edu/~liub/NSF/PSC-IIS-0307239.html

PU LEarning: cosine-rocchio PU algorithm, Li et al 2010 EMNLP
* Use cosine similarity and KNN (rochio classification) to filter likely false negatives from negative class to construct a "reliable negatives" dataset. Train SVM to further filter data. rinse and repeat

Perplexity based PU learning (stephen wan)
* use perplexity to find |pos_class| most dissimilar unlabeled instances to positive class
* train classifier to filter

**Comment by dmarx** [comment]
relevance models - ~2001/2002

LM ranking, top N -> another model to extract salient terms -> add to query, repeat  
use KL divergence instead of perplexity
